tests/code/test_users.py

```diff
@@ -34,63 +34,109 @@ def test_isEligibleForProgram():
-    result = addRemoveInterest(rule, program_id, user)
-    assert result == 'Successfully added interest'
-
-    # test add and removing interest with different program id
-    rule = 'addInterest'
+def test_addUserInterest():
+    username = "ramsayb2"
+    program_id = 2
+
+    # test adding interest for different users
+    result = addUserInterest(program_id, username)
+    assert result
+
+
+    username = "khatts"
+    result = addUserInterest(program_id, username)
+    assert result
+
+    # test adding interest with different program id
     program_id = 3
-    result = addRemoveInterest(rule, program_id, user)
-    assert result == 'Successfully added interest'
+    result = addUserInterest(program_id, username)
+    assert result
+
+    # test adding interest for user that does not exist
+    username = "jarjug"
+    program_id = 3
+    with pytest.raises(IntegrityError):
+        result = addUserInterest(program_id, username)
+        assert result
+
+    # test adding interest for program_id that does not exist
+    username = "khatts"
+    program_id = 45
+    with pytest.raises(IntegrityError):
+        result = addUserInterest(program_id, username)
+        assert result
+
+
+
 
-    rule = 'deleteInterest'
-    result = addRemoveInterest(rule, program_id, user)
-    assert result == 'Successfully removed interest'
 
 @pytest.mark.integration
-def test_addRemoveInvalidInterest():
+def test_removeUserInterestt():
+
+    #test for removing interest
+    username = "ramsayb2"
+    program_id = 2
+    result = removeUserInterest(program_id, username)
+    assert result ==  True
 
+    username = "khatts"
+    program_id = 2
+    result = removeUserInterest(program_id, username)
+    assert result == True
+
+    #test removing interest with different program id
+    program_id = 3
+    result = removeUserInterest(program_id, username)
+    assert result == True
+
+    # test adding interest for user that does not exist
+    username = "jarjug"
     program_id = 3
+    with pytest.raises(IntegrityError):
+        result = addUserInterest(program_id, username)
+        assert result == True
 
-    #test for user that doesn't exist
-    user = "al;skfjelh"
-    rule = 'addInterest'
+    # test adding interest for program_id that does not exist
+    username = "khatts"
+    program_id = 45
     with pytest.raises(IntegrityError):
-        result = addRemoveInterest(rule, program_id, user)
-        result == "Successfully added interest"
-
-    #test removing interest that doesn't exist
-    user = "lamichhanes2"
-    rule = 'deleteInterest'
-    program_id = 1
-    result = addRemoveInterest(rule, program_id, user)
-    assert result == "This interest does not exist"
-
-    # test for incorrect rule
-    rule = "lkejfiv"
-    result = addRemoveInterest(rule, program_id, user)
-    assert result == None
+        result = addUserInterest(program_id, username)
+        assert result == True
+
+
+@pytest.mark.integration
+def test_banUser():
```

Comment:
There's no failure in here. What happens if we failed to ban the user? If the answer is there's an exception, that's fine, but verify that in the test.